pkg/apis/configuration/v1alpha1/types.go

```diff
@@ -31,6 +31,7 @@ type Upstream struct {
 	LBMethod            string `json:"lb-method"`
 	FailTimeout         string `json:"fail-timeout"`
 	MaxFails            *int   `json:"max-fails"`
+	Keepalive           *int64 `json:"keepalive"`
```

Comment:
I think there's no need of using a pointer here. I would say `0` means keepalive is disabled (by default is disabled in the ConfigMap Key, so cfgm will be 0) Check https://github.com/nginxinc/kubernetes-ingress/blob/master/docs/configmap-and-annotations.md the `keepalive` ConfigMap Key

What do you think?

Reply:
yeah that makes sense

Reply:
I think we need to use a pointer here. Consider the case if the keeaplive is set in the ConfigMap, and the user wants to disable it. In that case, they can use:
```yaml
name: tea
service: tea-svc
port: 80
keepalive: 0
```
However, if you don't have keepalive as a pointer, the following yaml will also disable keepalives:
```yaml
name: tea
service: tea-svc
port: 80
```

Reply:
If we allow the user to disable it by setting it to 0 then yes a pointer is required @Dean-Coakley